### ISSUE TITLE
flowify_server(dev_server) start failed

### ISSUE THREAD

**Issue by Is4b3lla3** [equinor/flowify-workflows-server#49]
use command： `docker compose up -d`
<img width="1506" alt="image" src="https://user-images.githubusercontent.com/37106292/218320151-e6915dda-b9d7-4a3c-82bb-c04782ff1850.png">
- log
```
=====================================================================
Kind cluster exist, getting kubeconfig from cluster
Modifying Kubernetes config to point to Kind master node
=====================================================================

./flowify_server_runner.sh: line 14: [: -neq: binary operator expected

=====================================================================
Deploying flowify server
=====================================================================

time="2023-02-12T15:25:29Z" level=panic msg="can't load json schema, jsonschema file:///go/src/github.com/equinor/flowify-workflows-server/component.schema.json compilation failed: Get \"https://raw.githubusercontent.com/kubernetes/kubernetes/v1.21.2/api/openapi-spec/swagger.json\": dial tcp 0.0.0.0:443: connect: connection refused"
panic: (*logrus.Entry) 0x40001fd180

goroutine 1 [running]:
github.com/sirupsen/logrus.(*Entry).log(0x40001fd110, 0x0, {0x4000482360, 0x11e})
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:260 +0x4ac
github.com/sirupsen/logrus.(*Entry).Log(0x40001fd110, 0x0, {0x400059ea68?, 0x1?, 0x1?})
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:304 +0x64
github.com/sirupsen/logrus.(*Entry).Logf(0x40001fd110, 0x0, {0x19ba97b?, 0x1d393e0?}, {0x400059ec58?, 0x40000e7180?, 0x40002a7680?})
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/entry.go:349 +0x8c
github.com/sirupsen/logrus.(*Logger).Logf(0x40001b0000, 0x0, {0x19ba97b, 0x1a}, {0x400059ec58, 0x1, 0x1})
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/logger.go:154 +0x78
github.com/sirupsen/logrus.(*Logger).Panicf(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/logger.go:195
github.com/sirupsen/logrus.Panicf(...)
	/go/pkg/mod/github.com/sirupsen/logrus@v1.9.0/exported.go:219
github.com/equinor/flowify-workflows-server/models.init.0()
	/go/src/github.com/equinor/flowify-workflows-server/models/models.go:117 +0x5c4
```

**Comment by Is4b3lla3** [comment]
Due to network reasons, the above errors were caused. I increased the env variables of the http and https proxies, and the dev_server was able to start, but a new error was caused.
- log
```
INFO[0001] Starting process with pid 1038
INFO[0001] Setting global loglevel                       Loglevel=info
INFO[0001] No service account detected, running locally
INFO[0001] Nil mongo client is passed so a new client will be created. It is good practice to share clients
INFO[0001] Connected to mongodb ({mongo test map[address:mongo_server credentials:SET_FROM_ENV port:27017]}), with db name test
INFO[0001] Nil mongo client is passed so a new client will be created. It is good practice to share clients
INFO[0001] Connected to mongodb ({mongo test map[address:mongo_server credentials:SET_FROM_ENV port:%!s(int=27017)]}), with db name test
W0213 08:14:22.359463    1038 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
I0213 08:14:22.359556    1038 trace.go:205] Trace[1828635626]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169 (13-Feb-2023 08:14:07.863) (total time: 14495ms):
Trace[1828635626]: ---"Objects listed" error:Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF 14494ms (08:14:22.359)
Trace[1828635626]: [14.495038422s] [14.495038422s] END
E0213 08:14:22.359603    1038 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
W0213 08:14:37.462971    1038 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
I0213 08:14:37.463095    1038 trace.go:205] Trace[157874035]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169 (13-Feb-2023 08:14:23.354) (total time: 14108ms):
Trace[157874035]: ---"Objects listed" error:Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF 14108ms (08:14:37.462)
Trace[157874035]: [14.108366465s] [14.108366465s] END
E0213 08:14:37.463236    1038 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
W0213 08:14:54.229241    1038 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
I0213 08:14:54.229321    1038 trace.go:205] Trace[1631573571]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169 (13-Feb-2023 08:14:39.906) (total time: 14322ms):
Trace[1631573571]: ---"Objects listed" error:Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF 14321ms (08:14:54.229)
Trace[1631573571]: [14.322010381s] [14.322010381s] END
E0213 08:14:54.229337    1038 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
W0213 08:15:11.407254    1038 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
I0213 08:15:11.407343    1038 trace.go:205] Trace[920378501]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169 (13-Feb-2023 08:14:58.037) (total time: 13370ms):
Trace[920378501]: ---"Objects listed" error:Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF 13369ms (08:15:11.407)
Trace[920378501]: [13.370000423s] [13.370000423s] END
E0213 08:15:11.407357    1038 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
W0213 08:15:32.790355    1038 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
I0213 08:15:32.790409    1038 trace.go:205] Trace[510596729]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169 (13-Feb-2023 08:15:19.025) (total time: 13763ms):
Trace[510596729]: ---"Objects listed" error:Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF 13763ms (08:15:32.790)
Trace[510596729]: [13.763992839s] [13.763992839s] END
E0213 08:15:32.790461    1038 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
W0213 08:16:14.294255    1038 reflector.go:424] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
I0213 08:16:14.294352    1038 trace.go:205] Trace[2008996277]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169 (13-Feb-2023 08:15:56.586) (total time: 17707ms):
Trace[2008996277]: ---"Objects listed" error:Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF 17707ms (08:16:14.294)
Trace[2008996277]: [17.707434009s] [17.707434009s] END
E0213 08:16:14.294416    1038 reflector.go:140] pkg/mod/k8s.io/client-go@v0.25.2/tools/cache/reflector.go:169: Failed to watch *v1.ConfigMap: failed to list *v1.ConfigMap: Get "https://cluster-control-plane:6443/api/v1/namespaces/argo/configmaps?limit=500&resourceVersion=0": EOF
```

**Comment by OleksSokolov** [comment]
Hello, @Is4b3lla3 thanks for reaching out. 
I tried to reproduce the issue:
 `docker compose up -d` and everything works correctly.
 
 ===========
 Output:
 
osoko@T7V3WGG95C dev % docker compose up -d
[+] Running 23/24
 ⠿ server Warning                                                                                                                                                                                                1.9s
 ⠿ mongo Pulled                                                                                                                                                                                                 43.0s
   ⠿ 70cf24b16239 Pull complete                                                                                                                                                                                 17.3s
   ⠿ 197e79a59399 Pull complete                                                                                                                                                                                 17.4s
   ⠿ 1a54d1916136 Pull complete                                                                                                                                                                                 18.8s
   ⠿ 8547f0e899b2 Pull complete                                                                                                                                                                                 18.9s
   ⠿ 315a8963db2b Pull complete                                                                                                                                                                                 18.9s
   ⠿ 61b8bff0c93b Pull complete                                                                                                                                                                                 19.0s
   ⠿ 51d90acd9b0e Pull complete                                                                                                                                                                                 19.3s
   ⠿ c9aebc006dc1 Pull complete                                                                                                                                                                                 40.5s
   ⠿ 7e43614101f4 Pull complete                                                                                                                                                                                 40.6s
 ⠿ cluster Pulled                                                                                                                                                                                               22.2s
   ⠿ 213ec9aee27d Pull complete                                                                                                                                                                                  1.0s
   ⠿ 991cda7188e2 Pull complete                                                                                                                                                                                 18.3s
   ⠿ 4f4fb700ef54 Pull complete                                                                                                                                                                                 18.4s
   ⠿ d0b3f7ee2e27 Pull complete                                                                                                                                                                                 18.6s
   ⠿ 699405061404 Pull complete                                                                                                                                                                                 18.7s
   ⠿ 0eaa3840805a Pull complete                                                                                                                                                                                 20.4s
   ⠿ ccdf4d4a3a34 Pull complete                                                                                                                                                                                 20.4s
   ⠿ bad73002a173 Pull complete                                                                                                                                                                                 20.4s
   ⠿ 7120428208b5 Pull complete                                                                                                                                                                                 20.5s
   ⠿ beba9bae608e Pull complete                                                                                                                                                                                 20.5s
   ⠿ 2967f7eedcfb Pull complete                                                                                                                                                                                 20.6s
   ⠿ 636bc7d21cce Pull complete                                                                                                                                                                                 20.6s
[+] Building 29.6s (34/34) FINISHED
 => [internal] load build definition from Dockerfile.server                                                                                                                                                      0.0s
 => => transferring dockerfile: 1.39kB                                                                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/golang:1.18-alpine                                                                                                                                            1.4s
 => [base 1/2] FROM docker.io/library/golang:1.18-alpine@sha256:77f25981bd57e60a510165f3be89c901aec90453fd0f1c5a45691f6cb1528807                                                                                 0.0s
 => [internal] load build context                                                                                                                                                                                0.0s
 => => transferring context: 697.88kB                                                                                                                                                                            0.0s
 => CACHED [base 2/2] RUN apk add --no-cache     bash     binutils     curl     docker     gcc     git     jq     make     musl-dev     openssl     shadow                                                       0.0s
 => CACHED [buildbase  1/11] WORKDIR /root                                                                                                                                                                       0.0s
 => CACHED [buildbase  2/11] RUN curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/v0.15.0/kind-linux-amd64 &&     chmod +x ./kind &&     mv ./kind /usr/local/bin/kind                  0.0s
 => CACHED [buildbase  3/11] RUN mkdir -p /go/src/github.com/equinor/                                                                                                                                            0.0s
 => CACHED [buildbase  4/11] WORKDIR /go/src/github.com/equinor/flowify-workflows-server                                                                                                                         0.0s
 => CACHED [buildbase  5/11] COPY dev/flowify_server_runner.sh .                                                                                                                                                 0.0s
 => CACHED [buildbase  6/11] RUN chmod +x ./flowify_server_runner.sh                                                                                                                                             0.0s
 => CACHED [buildbase  7/11] COPY dev/kind_cluster_config_export.sh .                                                                                                                                            0.0s
 => CACHED [buildbase  8/11] RUN chmod +x ./kind_cluster_config_export.sh                                                                                                                                        0.0s
 => CACHED [buildbase  9/11] COPY go.mod .                                                                                                                                                                       0.0s
 => CACHED [buildbase 10/11] COPY go.sum .                                                                                                                                                                       0.0s
 => CACHED [buildbase 11/11] RUN go mod download                                                                                                                                                                 0.0s
 => CACHED [devbase 1/2] RUN go install github.com/jstemmer/go-junit-report@latest                                                                                                                               0.0s
 => CACHED [devbase 2/2] RUN go install github.com/jandelgado/gcov2lcov@latest                                                                                                                                   0.0s
 => CACHED [devserver  1/14] COPY apiserver ./apiserver                                                                                                                                                          0.0s
 => CACHED [devserver  2/14] COPY auth ./auth                                                                                                                                                                    0.0s
 => CACHED [devserver  3/14] COPY cmd ./cmd                                                                                                                                                                      0.0s
 => CACHED [devserver  4/14] COPY models ./models                                                                                                                                                                0.0s
 => CACHED [devserver  5/14] COPY pkg ./pkg                                                                                                                                                                      0.0s
 => [devserver  6/14] COPY rest ./rest                                                                                                                                                                           0.0s
 => [devserver  7/14] COPY storage ./storage                                                                                                                                                                     0.0s
 => [devserver  8/14] COPY transpiler ./transpiler                                                                                                                                                               0.0s
 => [devserver  9/14] COPY user ./user                                                                                                                                                                           0.0s
 => [devserver 10/14] COPY config.yml .                                                                                                                                                                          0.0s
 => [devserver 11/14] COPY main.go .                                                                                                                                                                             0.0s
 => [devserver 12/14] COPY Makefile .                                                                                                                                                                            0.0s
 => [devserver 13/14] COPY e2etest ./e2etest                                                                                                                                                                     0.0s
 => [devserver 14/14] RUN ["/bin/bash", "-c", "make server strip=1 flowify_git_sha=${FLOWIFY_GIT_SHA}"]                                                                                                         26.9s
 => exporting to image                                                                                                                                                                                           1.0s
 => => exporting layers                                                                                                                                                                                          1.0s
 => => writing image sha256:d40ae1c35e8edf80be009a2dca0d7eca3f5b05e106f6588e2248bca5309b3d15                                                                                                                     0.0s
 => => naming to docker.io/library/dev_server                                                                                                                                                                    0.0s
[+] Running 3/4
 ⠿ Container mongo_server                                                                                                                                 Started                                                0.3s
 ⠿ Container kind_cluster                                                                                                                                 Healthy                                               67.0s
 ⠹ cluster The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested                                                        0.0s
 ⠿ Container flowify_server
 
 ==================
 
<img width="1367" alt="Screenshot 2023-02-13 at 16 28 17" src="https://user-images.githubusercontent.com/121943814/218485847-1fc757cb-1e3a-4899-9972-d9c5a62eb965.png">

**Comment by OleksSokolov** [comment]
I would like to suggest removing the existing containers, images, and volumes (or `docker compose down` and `docker system prune` if it is possible). And try again.

**Comment by Is4b3lla3** [comment]
Thank you very much, I have solved this problem, the main reason is my own network problem inside docker.
I also ran into another problem:

The response to clicking the artifacts download button in the workflow results is 404, and there is no response to clicking the `fetch logs`. Is this my own problem or the function has not been developed yet?

<img width="1353" alt="image" src="https://user-images.githubusercontent.com/37106292/218691306-d8fe4398-4445-49a6-95a1-da2a6f388357.png">

Finally, I want to know when the form of k8s deployment can be added
